### PR TITLE
Add note regarding parent module containing use statement.

### LINKED ISSUE #40341: Suggest to move/copy use statement from parent into inline module declaration
Given the below Rust code, we can probably produce a more user friendly error message than today's "failed to resolve. Use of undeclared type or module `File`".

```rust
use std::fs::File;

mod foo {
    fn some_function(f: File) {}
}
```

It'd be nice to suggest something like "Possibly move `use std::fs::File` from outer scope into module `foo`". This will help beginners understand how to deal with the `File` use in the parent module not being in scope within the child module.

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
Fixes #40341.

### CODE REVIEW

File: src/librustc_resolve/diagnostics.rs [hunk 1]
```diff
@@ -838,6 +838,31 @@ trait Foo {
 
 fn foo<T>(x: T) {} // ok!
 ```
+
+Another case that causes this error is when a type is imported into a parent
+module. To fix this, you can follow the suggestion and use File directly or
+`use super::File;` which will import the types from the parent namespace. An
+example that causes this error is below:
+
+```
```
Comment (kennytm) [inline src/librustc_resolve/diagnostics.rs]: needs to be ````` ```ignore `````

File: src/librustc_resolve/diagnostics.rs [hunk 2]
```diff
@@ -838,6 +838,31 @@ trait Foo {
...
+    fn some_function(f: File) {}
+}
+```
+
+```
+use std::fs::File;
+
+mod foo {
+    // either
+    use super::File;
```
Comment (kennytm) [inline src/librustc_resolve/diagnostics.rs]: Currently CI failed with 

```
[00:55:49] ---- /checkout/obj/build/x86_64-unknown-linux-gnu/test/error-index.md - Rust_Compiler_Error_Index (line 6486) stdout ----
[00:55:49] 	error[E0432]: unresolved import `super::File`
[00:55:49]  --> <anon>:6:9
[00:55:49]   |
[00:55:49] 6 |     use super::File;
[00:55:49]   |         ^^^^^^^^^^^ no `File` in the root
```

Dunno why, probably workaround it by adding a hidden `# pub` above the `use std::fs::File`?

Comment (Mark-Simulacrum) [inline src/librustc_resolve/diagnostics.rs]: The problem here was that rustdoc auto-injects `fn main` surrounding this, so it was correct -- there was no `File` in root. I inserted an empty commented fn main to workaround this, so we should be fine now.

**Pull request merged**